components/style/gecko_conversions.rs

```diff
@@ -70,6 +70,29 @@ impl From<LengthOrPercentage> for nsStyleCoord_CalcValue {
     }
 }
 
+impl LengthOrPercentageOrAuto {
+    pub fn into_calc_value(&self) -> Option<nsStyleCoord_CalcValue> {
```

Comment:
Is "into" really the right name here? It seems like that would imply that the value is consumed, which it isn't here.

In general it seems like we want to impl From, but maybe that runs into coherence issues? I'd go with as_ or to_ for now.